dartagnan/src/main/java/com/dat3m/dartagnan/asserts/AssertNot.java

```diff
@@ -27,4 +31,11 @@ public BooleanFormula encode(SolverContext ctx) {
     public String toString() {
         return "!" + child.toString();
     }
+
+    @Override
+	public List<Register> getRegs() {
+		List<Register> regs = new ArrayList<>();
+		regs.addAll(child.getRegs());
+		return regs;
```

Comment:
you could also just return `child.getRegs` or if you really need a copy, you can do "return new ArrayList(child.getRegs())".